components/plugin_marketplace/marketplace_item/marketplace_item.jsx

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {FormattedMessage} from 'react-intl';
+
+import {Link} from 'react-router-dom';
+
+import LoadingWrapper from 'components/widgets/loading/loading_wrapper.tsx';
+import ConfirmModal from 'components/confirm_modal.jsx';
+import PluginIcon from 'components/widgets/icons/plugin_icon.jsx';
+
+import {trackEvent} from 'actions/diagnostics_actions.jsx';
+import {localizeMessage} from 'utils/utils';
+
+export default class MarketplaceItem extends React.Component {
+        static propTypes = {
+            id: PropTypes.string.isRequired,
+            name: PropTypes.string.isRequired,
+            description: PropTypes.string.isRequired,
+            version: PropTypes.string.isRequired,
+            downloadUrl: PropTypes.string,
+            homepageUrl: PropTypes.string,
+            iconData: PropTypes.string,
+            installed: PropTypes.bool.isRequired,
+            onConfigure: PropTypes.func.isRequired,
+            onInstalled: PropTypes.func.isRequired,
+            actions: PropTypes.shape({
+                installPluginFromUrl: PropTypes.func.isRequired,
+            }).isRequired,
+        };
+
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                installed: this.props.installed,
```

Comment:
nit: We generally omit `this` and use `props.installed` in constructors

Reply:
Is there a reason that we're setting `installed` in state? Won't the prop get updated when the plugin is installed? Also what happens when I'm looking at this page and someone else installs the plugin? I don't see the state being updated from the prop anywhere which could cause inconsistencies and lead to errors.

Reply:
> Is there a reason that we're setting installed in state? Won't the prop get updated when the plugin is installed? 

Yes it will update. Left over from initial dev it was used to store the local state of the component when it clicked to be installed but now it does synch with the server after it completes. Removing. 

>  Also what happens when I'm looking at this page and someone else installs the plugin? 

If you are looking at this page and take no action (i.e. not installing or searching anything that would trigger a refresh) you could have the situation that that the server has a plugin installed that is listed as not installed. However, if they do click on install it will prompt the user if they want to replace it. It won't cause an error but I agree it can be inconsistent and confusing. 

We need to improve this experience (and probably the same with System Console/Plugins Management) by being able to refresh on server changes.